lesson1-2

fav77 opened the pull request:
lesson1: comp
lesson2.1:comp
lesson2.2:comp

fav77 committed:
lesson1: comp
lesson2.1:comp
lesson2.2:comp

fav77 committed:
lesson1: comp
lesson2.1:comp (changed)
lesson2.2:comp

belyaev-mikhail commented:
В ваших решениях присутствуют ошибки. Их следует исправить.

belyaev-mikhail commented:
recheck assignment

belyaev-mikhail commented on src/lesson1/task1/Simple.kt:
```diff
@@ -60,31 +60,32 @@ fun seconds(hours: Int, minutes: Int, seconds: Int): Int = TODO()
  * 1 сажень = 3 аршина = 48 вершков, 1 вершок = 4.445 см.
  */
-fun lengthInMeters(sagenes: Int, arshins: Int, vershoks: Int): Double = TODO()
+fun lengthInMeters(sagenes: Int, arshins: Int, vershoks: Int): Double = (sagenes * 48 + arshins * 16 + vershoks)* 4.445/100
```
Длинные сроки, не влезающие в экран, в коде недопустимы, их следует переносить.

belyaev-mikhail commented on src/lesson1/task1/Simple.kt:
```diff
@@ -60,31 +60,32 @@ fun seconds(hours: Int, minutes: Int, seconds: Int): Int = TODO()
  * Вывести значение того же угла в радианах (например, 0.63256).
  */
-fun angleInRadian(grad: Int, min: Int, sec: Int): Double = TODO()
+fun angleInRadian(grad: Int, min: Int, sec: Int): Double = (grad+min/60.0+sec/(60.0*60))/(180/PI)
```
Длинные сроки, не влезающие в экран, в коде недопустимы, их следует переносить.

belyaev-mikhail commented on src/lesson1/task1/Simple.kt:
```diff
@@ -60,31 +60,32 @@ fun seconds(hours: Int, minutes: Int, seconds: Int): Int = TODO()
  * Определить третью цифру справа в этом числе (в данном случае 8).
  */
-fun thirdDigit(number: Int): Int = TODO()
+fun thirdDigit(number: Int): Int {
```
Следует корректно форматировать код. Закрывающая скобка функции ставится на отдельной строке.

belyaev-mikhail commented on src/lesson1/task1/Simple.kt:
```diff
@@ -102,12 +107,19 @@ fun travelMinutes(hoursDepart: Int, minutesDepart: Int, hoursArrive: Int, minute
+    val a = number%10
+    val b = (number/10)%10
+    val c = (number/100)%10
+    return a*100+b*10+c }
```
Аналогично

belyaev-mikhail commented on src/lesson2/task1/IfElse.kt:
```diff
@@ -44,7 +49,13 @@ fun ageDescription(age: Int): String = TODO()
+                   t3: Double, v3: Double): Double {
+    val hS = (t1*v1 + t2*v2 + t3*v3)/2
+    if (hS <= v1*t1) return hS/v1
+    if (hS <= (v1*t1 + v2*t2)) return t1 + ((hS - v1 * t1) / v2)
```
Следует придерживаться _единого_ стиля форматирования кода. Если вы ставите пробелы до и после знаков, то следует их ставить **везде**. Если не ставите &mdash; их не следует ставить **нигде**.

belyaev-mikhail commented on src/lesson2/task1/IfElse.kt:
```diff
@@ -89,4 +118,13 @@ fun triangleKind(a: Double, b: Double, c: Double): Int = TODO()
+        return b - c
+    if (b >= c && c >= a && b >= d)
+        return d - c
+                                   else return -1 }
```
См. комментарий про форматирование

belyaev-mikhail commented on src/lesson2/task2/Logical.kt:
```diff
@@ -46,4 +61,8 @@ fun circleInside(x1: Double, y1: Double, r1: Double,
+    if (s>=a && (r>=b || r>=c)) return true
+    if (s>=b && (r>=a || r>=c)) return true
+    if (s>=c && (r>=b || r>=a)) return true
+    return false }
```
См. комментарий про форматирование

fav77 committed:
lesson1: comp
lesson2.1:comp (changed)
lesson2.2:comp
lesson3:(half-complete)

SidorenkoNikita committed:
Исправлены ошибки и оформление lesson2.1

SidorenkoNikita committed:
Исправлены ошибки и оформление lesson2.1

belyaev-mikhail commented:
Почему у вас в одном PR коммиты от разных людей?
Как говорят учителя в школе, "оценку мы тоже на двоих делить будем"?

fav77 committed:
Исправлены ошибки и оформление lesson2.1

belyaev-mikhail commented on src/lesson3/task1/Loop.kt:
```diff
@@ -80,14 +96,22 @@ fun lcm(m: Int, n: Int): Int = TODO()
  */
-fun maxDivisor(n: Int): Int = TODO()
+fun maxDivisor(n: Int): Int {
+        var div = n - 1
```
Отступы в коде должны быть консистентными

belyaev-mikhail commented on src/lesson3/task1/Loop.kt:
```diff
@@ -80,14 +96,22 @@ fun lcm(m: Int, n: Int): Int = TODO()
  */
-fun minDivisor(n: Int): Int = TODO()
+fun minDivisor(n: Int): Int {
+        var div = 2
```
Отступы в коде должны быть консистентными

belyaev-mikhail commented on src/lesson3/task1/Loop.kt:
```diff
@@ -57,15 +57,31 @@ fun digitCountInNumber(n: Int, m: Int): Int =
+        result += prev
+        prev = result - prev
+    }
+                        return result
```
Отступы в коде должны быть консистентными

belyaev-mikhail commented:
Исправьте, пожалуйста, замечания по оформлению.

fav77 committed:
lesson1: comp
lesson2.1:comp (changed)
lesson2.2:comp
lesson3:(half-complete)
"Отступы в коде должны быть консистентными" Исправлено

fav77 committed:
lesson1: comp
lesson2.1:comp (changed)
lesson2.2:comp
lesson3:(half-complete)
"Отступы в коде должны быть консистентными" Исправлено

fav77 committed:
Merge remote-tracking branch 'origin/master'

# Conflicts:
#	src/lesson2/task1/IfElse.kt

fav77 committed:
Исправлены ошибки

fav77 commented:
Прошу прощение за такое количество работ ("Push & Commit") . Это произошло из-за зависания программы.

belyaev-mikhail commented:
recheck all

belyaev-mikhail commented on src/lesson2/task2/Logical.kt:
```diff
@@ -35,7 +46,11 @@ fun queenThreatens(x1: Int, y1: Int, x2: Int, y2: Int): Boolean = TODO()
 fun circleInside(x1: Double, y1: Double, r1: Double,
-                 x2: Double, y2: Double, r2: Double): Boolean = TODO()
+                 x2: Double, y2: Double, r2: Double): Boolean {
+    if (((Math.sqrt((x1 - x2) * (x1 - x2) + (y1 - y2) * (y1 - y2))) + r1) <= r2) return true
```
Конструкция `if (x) return true else return false` лишена смысла, т.к. ничем не отличается от просто `return x`
